pandas-stubs/_typing.pyi

```diff
@@ -64,21 +64,32 @@ DtypeObj = Union[np.dtype[np.generic], ExtensionDtype]
 AnyStr_cov = TypeVar("AnyStr_cov", str, bytes, covariant=True)
 AnyStr_con = TypeVar("AnyStr_con", str, bytes, contravariant=True)
 
-class BaseBuffer(Protocol): ...
-class ReadBuffer(BaseBuffer, Protocol[AnyStr_cov]): ...
-class WriteBuffer(BaseBuffer, Protocol[AnyStr_cov]): ...
+class BaseBuffer(Protocol):
+    @property
+    def mode(self) -> str: ...
+    def seek(self, __offset: int, __whence: int = ...) -> int: ...
+    def seekable(self) -> bool: ...
+    def tell(self) -> int: ...
+
+class ReadBuffer(BaseBuffer, Protocol[AnyStr_cov]):
+    def read(self, __n: int = ...) -> AnyStr_cov: ...
+
+class WriteBuffer(BaseBuffer, Protocol[AnyStr_con]):
+    def write(self, __b: AnyStr_con) -> Any: ...
+    def flush(self) -> Any: ...
 
 class ReadPickleBuffer(ReadBuffer[bytes], Protocol):
-    def readline(self, size: int | None = ...) -> bytes: ...
+    def readline(self) -> bytes: ...
```

Comment:
All the buffers are used as input arguments (not as return values), so we can remove not needed optional arguments to be compatible with a wider class of IO functions that might or might not implement optional keywords.